modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/events/CookiesPreAction.java

```diff
@@ -0,0 +1,274 @@
+import com.liferay.portal.kernel.events.LifecycleAction;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Carol Alonso
+ */
+@Component(
+	property = "key=servlet.service.events.pre", service = LifecycleAction.class
+)
+public class CookiesPreAction extends Action {
+
+	@Override
+	public void run(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		try {
+			_run(httpServletRequest, httpServletResponse);
+		}
+		catch (Exception exception) {
+			_log.error(exception);
+		}
+	}
+
+	private void _addCookie(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse, Cookie cookie) {
+
+		cookie.setPath("/");
+		cookie.setVersion(0);
+
+		if (cookie.getMaxAge() != 0) {
+			cookie.setMaxAge(365 * 24 * 60 * 60);
+		}
+
+		CookiesManagerUtil.addCookie(
+			CookiesConstants.CONSENT_TYPE_NECESSARY, cookie, httpServletRequest,
+			httpServletResponse);
+	}
+
+	private void _expireCookie(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse, String name) {
+
+		Cookie cookie = new Cookie(name, null);
+
+		cookie.setMaxAge(0);
+
+		_addCookie(httpServletRequest, httpServletResponse, cookie);
+	}
+
+	private void _expireNecessaryCookies(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		for (String necessaryCookieName : CookiesPreAction._NECESSARY_COOKIES) {
+			_expireCookie(
+				httpServletRequest, httpServletResponse, necessaryCookieName);
+		}
+	}
+
+	private void _expireOptionalCookies(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		for (String optionalCookieName : CookiesPreAction._OPTIONAL_COOKIES) {
+			_expireCookie(
+				httpServletRequest, httpServletResponse, optionalCookieName);
+		}
+	}
+
+	private void _expireUserConsentConfiguredCookie(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		_expireCookie(
+			httpServletRequest, httpServletResponse,
+			CookiesConstants.NAME_USER_CONSENT_CONFIGURED);
+	}
+
+	private Map<String, String> _parseCookieMap(Cookie[] cookies) {
+		Map<String, String> cookieMap = new HashMap<>();
+
+		if (cookies != null) {
+			for (Cookie cookie : cookies) {
+				String cookieName = cookie.getName();
+
+				if (cookieName.startsWith("CONSENT_TYPE_") ||
```

Comment:
Just a silly thing but I think it is preferred to first check the equals condition and then the startsWith (alphabetical order...)